### ISSUE TITLE
Intl.DisplayNames should throw a TypeError when "type" option is not specified, but test262 does not assume so

### ISSUE THREAD

**Issue by Constellation** [tc39/test262#2682]
The draft spec describes that if "type" option is undefined we will throw a TypeError.
https://tc39.es/proposal-intl-displaynames/#sec-Intl.DisplayNames
> 13. Let type be ? GetOption(options, "type", "string", « "language", "region", "script", "currency" », undefined).
> 14. If type is undefined, throw a TypeError exception.

So,
```
new Intl.DisplayNames()
```
should throw a TypeError.

But, tests in test262 disagree with this. Here is an example of tests.
https://github.com/tc39/test262/blob/master/test/intl402/DisplayNames/ctor-custom-get-prototype-poison-throws.js#L43
https://github.com/tc39/test262/blob/master/test/intl402/DisplayNames/ctor-custom-prototype.js#L35
https://github.com/tc39/test262/blob/master/test/intl402/DisplayNames/ctor-default-prototype.js#L19

I observed that, when exposing `Intl.DisplayNames`, tests using `testWithIntlConstructors` start failing since `testWithIntlConstructors` lists `DisplayNames` and tests using `testWithIntlConstructors` are constructing DisplayNames instances without passing `{ type: ... }` option.

I think that this is spec issue and we should offer some fallback like `"language"`, is it right? Or should we fix test262 tests?

**Comment by sffc** [comment]
This was a recent change.  The Test262 probably still needs to be updated. https://github.com/tc39/proposal-intl-displaynames/issues/73

@FrankYFTang @ryzokuken

**Comment by ryzokuken** [comment]
@sffc should we wait for the discussion at https://github.com/tc39/proposal-intl-displaynames/issues/73#issuecomment-651427251 to reach a conclusion before making an update here?

**Comment by sffc** [comment]
Yeah, let's discuss this issue at the 402 meeting next week.  I think it's likely that we'll stick with the TypeError, but we might as well wait a week before doing extra work that might not end up being necessary.

**Comment by sffc** [comment]
@FrankYFTang Does 3c89cde3e02849cd9f06a1a8344b45af56c5e142 fully resolve @Constellation's issue in the OP?
